### PR TITLE
Update accessibility contrast test coverage

### PR DESCRIPTION
**Pull request by ankur2136**
Update the accessibility test to cover all Elements with same semantic label to get better coverage on contrast test.

#109465 Improve flutter contrast test coverage

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by chunhtai — APPROVED** [review]
LGTM

**Comment by chunhtai** [comment]
CC @goderbauer for secondary review

File: packages/flutter_test/lib/src/accessibility.dart [hunk 1]
```diff
@@ -331,50 +331,52 @@ class MinimumTextContrastGuideline extends AccessibilityGuideline {
...
+    return result;
+  }
 
+  Future<Evaluation> _evaluateElement(
+      SemanticsNode node,
+      Element element,
+      WidgetTester tester,
+      ui.Image image,
+      ByteData byteData,
+      ) async {
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:348]: nit: indentation:

```suggestion
    SemanticsNode node,
    Element element,
    WidgetTester tester,
    ui.Image image,
    ByteData byteData,
  ) async {
```

File: packages/flutter_test/lib/src/accessibility.dart [hunk 2]
```diff
@@ -331,50 +331,52 @@ class MinimumTextContrastGuideline extends AccessibilityGuideline {
...
+    const Evaluation result = Evaluation.pass();
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:349]: What is this extra const variable for? Can you just use `const Evaluation.pass()` in place of `result` below to clear up the code?

**Review by goderbauer — APPROVED** [review]
LGTM after comments are resolved.

File: packages/flutter_test/lib/src/accessibility.dart (hunk 2 above)
Comment (ankur2136) [reply packages/flutter_test/lib/src/accessibility.dart]: Updated.

File: packages/flutter_test/lib/src/accessibility.dart (hunk 1 above)
Comment (ankur2136) [reply packages/flutter_test/lib/src/accessibility.dart]: Updated.

File: packages/flutter_test/lib/src/accessibility.dart [hunk 3]
```diff
@@ -331,61 +331,62 @@ class MinimumTextContrastGuideline extends AccessibilityGuideline {
...
+    return result;
+  }
 
+  Future<Evaluation> _evaluateElement(
+    SemanticsNode node,
+    Element element,
+    WidgetTester tester,
+    ui.Image image,
+    ByteData byteData,
+    ) async {
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:348]: unindent this line by two spaces.

Comment (ankur2136) [reply packages/flutter_test/lib/src/accessibility.dart]: Done

**Comment by jonahwilliams** [comment]
Won't this pick up text that has been excluded from semantics?

**Comment by ankur2136** [comment]
> Won't this pick up text that has been excluded from semantics?

Hasn't happened in any test in GP3. The fix has identified 470 tests however where we needed to manually handle failures.

**Comment by jonahwilliams** [comment]
Also, if we're going to look at every text element anyway - then why bother looking at the semantics nodes at all. Shouldn't we just traverse the element tree to gather every text element?

**Review by goderbauer — CHANGES_REQUESTED** [review]
>> Won't this pick up text that has been excluded from semantics?
>
> Hasn't happened in any test in GP3. The fix has identified 470 tests however where we needed to manually handle failures.

Can you add a test case to verify that this isn't happening, please?

**Comment by goderbauer** [comment]
@jonahwilliams has a good point regarding element traversal...

**Comment by ankur2136** [comment]
> > > Won't this pick up text that has been excluded from semantics?
> > 
> > 
> > Hasn't happened in any test in GP3. The fix has identified 470 tests however where we needed to manually handle failures.
> 
> Can you add a test case to verify that this isn't happening, please?

Added another test with a contrast failing Text wrapped in ExcludeSemantics and a duplicate Text with contrast within guidelines.

**Comment by goderbauer** [comment]
You probably have to rebase this with the latest master to get tests to pass again.

**Comment by ankur2136** [comment]
Closed this since it caused a bad merge with master and somehow went ahead of flutter/master. 

Created a new PR in https://github.com/flutter/flutter/pull/109784

**Pull request closed without merging**